File: src/flipAndInvertImage.c
#include <stdlib.h>

int **flipAndInvertImage(int **image, int imageSize, int *imageColSize,
                         int *returnSize, int **returnColumnSizes) {
    *returnSize = imageSize;
    *returnColumnSizes = (int *)malloc(imageSize * sizeof(int));
    int columns = *imageColSize;
    int **new_image = (int **)malloc(imageSize * sizeof(int *));
    for (int i = 0; i < imageSize; i++) {
        (*returnColumnSizes)[i] = columns;
        new_image[i] = (int *)malloc(columns * sizeof(int));
        int l = 0, r = columns - 1;
        while (l <= r) {
            new_image[i][l] = image[i][r] ^ 1;
            new_image[i][r] = image[i][l] ^ 1;
            l++;
            r--;
        }
    }
    return new_image;
}

File: src/flipAndInvertImage.py
class Solution:
    def flipAndInvertImage(self, image: list[list[int]]) -> list[list[int]]:
        columns = len(image[0])
        always_r = columns - 1
        updated_image = []
        for row in image:
            updated_row = [0] * columns
            l = 0
            r = always_r
            while l <= r:
                updated_row[l] = int(not row[r])
                updated_row[r] = int(not row[l])
                l += 1
                r -= 1
            updated_image.append(updated_row)
        return updated_image
